Sources/Markdown/Base/Markup.swift

```diff
@@ -204,7 +204,7 @@ extension Markup {
                 childId: firstChildID.childId + siblingSubtreeCount
             )
             
-            childMetadata = MarkupMetadata(id: childID, indexInParent: indexInParent + position)
+            childMetadata = MarkupMetadata(id: childID, indexInParent: position)
```

Comment:
The `indexInParent` value of a child is unrelated to the parent's `indexInParent`. I missed this initially because tests were only checking for children of the first item where `indexInParent` would be 0. This PR includes a new test that asserts `Markup.child(at:)` returns correct values for children of nested items as well.

Reply:
Heh, nice catch. Sorry I also missed that.